### ISSUE TITLE
Revamp connection settings

### ISSUE THREAD

**Issue by nataliekwong** [airbytehq/airbyte#29896]
We know the connection / replication settings are a sore spot for users. Over time, we've continued to add more settings without logically placing them in the flow to ensure users understand what's important. 

Wireframe: https://whimsical.com/connection-settings-RmUcF7ZUCVfL6aLaCSrsRi

Proposed changes to connection setup:

Split the connection setup into two steps:	
- Connection Setup (same as is, with reduced settings) to kick off the first sync
- Connection Success screen (new screen, with non-essential settings)
- Introduce “Advanced Settings” in Connection Setup intended only for power users that serve a limited number of use cases

Proposed changes to connection edit:
Connection Settings includes:
- Connection Name [same]
- Sync Frequency [new]
- Detect and Propagate Schema Changes [new]
- Schema update notifications [same]
- Track Change History (tbd - new)

Replication Settings includes:

- Streams List (PK, Cursor, Columns) [same]
- Namespace [same, but not modifiable after create]
- Advanced Settings [new]
- Data Residency [new] (should this be editable?)
- Destination Stream Prefix [new, but not modifiable after create]

**Comment by tealjulia** [comment]
refining notes:
- move the sync frequency to the "setup connection" step near "destination folder"
    - change the button text to "save connection" instead of "start sync" if schedule is manual or cron, set to "Save and sync" if other
- advanced settings inline with the footer of the table (where the buttons are)


asides:
- maybe the destination metadata can tell us (a) what a better term for namespace is and (b) whether or not the destination namespace _can_ be overridden

**Comment by nora-airbyte** [comment]
[MOCKS](https://www.figma.com/file/cBBHFnMQyHCtRZM6uWIkbA/Simplify-connection-settings?type=design&node-id=15%3A1501&mode=design&t=j5eFkD64zBrmcLXy-1)

@chandlerprall Here's a first pass on these mocks. A few differences here from the wireframes - let's go over them together

**Comment by nora-airbyte** [comment]
[New mocks](https://www.figma.com/file/cBBHFnMQyHCtRZM6uWIkbA/Simplify-connection-settings?type=design&node-id=154%3A3317&mode=design&t=GPg2mk2LrcUr9bxh-1) @nataliekwong. I think the only unresolved thing here is sync mode, correct? Not sure where we netted out on having it/getting rid of it.

**Comment by nataliekwong** [comment]
Yeah, I'd still air on keeping it as a configuration option, and at least try through any doubts of edge cases and how we could try to handle them rather than taking off the table.

I think the method Lake suggested was:
- Default is chosen (Incremental | Deduped + Append)
- Anything that is still the default option is greyed out
- Anything deviating from the default which is changed by the user is not greyed out
